services/gitdiff/gitdiff.go

```diff
@@ -577,9 +580,25 @@ parsingLoop:
 				break curFileLoop
 			case strings.HasPrefix(line, "old mode ") ||
 				strings.HasPrefix(line, "new mode "):
+
+				if strings.HasPrefix(line, "old mode ") {
+					r := regexp.MustCompile(`old mode (\d{6})`)
+					matched := r.FindStringSubmatch(line)
+					curFile.OldMode = matched[1]
+				}
+				if strings.HasPrefix(line, "new mode ") {
+					r := regexp.MustCompile(`new mode (\d{6})`)
+					matched := r.FindStringSubmatch(line)
+					curFile.Mode = matched[1]
+				}
+
 				if strings.HasSuffix(line, " 160000\n") {
 					curFile.IsSubmodule = true
 				}
+			case strings.HasPrefix(line, "new file mode "):
+				r := regexp.MustCompile(`new file mode (\d{6})`)
```

Comment:
These regexes should probably be extracted into a "constant".
This way, they are only computed once instead of once per diff file.

Reply:
IMO maybe it could use a string search (faster?), no need to use regexp.

Reply:
That's also an option…

Reply:
Done